src/Microsoft.Azure.SignalR.Common/ServiceConnections/MultiEndpointServiceConnectionContainer.cs

```diff
@@ -33,23 +39,21 @@ internal class MultiEndpointServiceConnectionContainer : IServiceConnectionConta
             if (generator == null)
             {
                 throw new ArgumentNullException(nameof(generator));
-            }
-
+            }
+
+            _router = router ?? throw new ArgumentNullException(nameof(router));
             _logger = loggerFactory?.CreateLogger<MultiEndpointServiceConnectionContainer>() ?? throw new ArgumentNullException(nameof(loggerFactory));
 
             // provides a copy to the endpoint per container
             _endpoints = endpointManager.GetEndpoints(hub);
 
-            if (_endpoints.Count == 1)
-            {
-                _inner = generator(_endpoints[0]);
-            }
-            else
-            {
-                // router is required when endpoints > 1
-                _router = router ?? throw new ArgumentNullException(nameof(router));
-                ConnectionContainers = _endpoints.ToDictionary(s => (ServiceEndpoint)s, s => generator(s));
-            }
+            foreach (var endpoint in _endpoints)
+            {
+                _connectionContainers[endpoint] = generator(endpoint);
+            }
+
+            // assign first element to _inner for quick access in single endpoint case
+            _inner = _connectionContainers.First().Value;
```

Comment:
`First()` -> `[0]`

Reply:
it throws when _endpoints.Length == 0

Reply:
Offline synced to remove _inner to work in same path.